lib/matplotlib/cbook/__init__.py

```diff
@@ -1397,8 +1397,11 @@ def _reshape_2D(X, name):
         # 1D array of scalars: directly return it.
         return [X]
     elif X.ndim in [1, 2]:
-        # 2D array, or 1D array of iterables: flatten them first.
-        return [np.reshape(x, -1) for x in X]
+        if hasattr(X[0], '__len__'):
```

Comment:
Or just replace the `X.dtype.type != np.object_` check above by `not hasattr(X[0], "__iter__")`/`not isinstance(X[0], collections.abc.Iterable)`?  That seems to match what the comments intended too.  (`__iter__` or `__len__`, I don't really care.)